src/Components/ServiceScene/Breakdowns/ByFrameRepeater.tsx

```diff
@@ -20,6 +21,7 @@ type FrameFilterCallback = (frame: DataFrame) => boolean;
 type FrameIterateCallback = (frames: DataFrame[], seriesIndex: number) => void;
 
 export class ByFrameRepeater extends SceneObjectBase<ByFrameRepeaterState> {
+  public changepointDetector = ChangepointDetector.defaultArgpcp();
```

Comment:
```suggestion
```

I'm now realising that the docs are really unclear about this, but this should be recreated afresh for each series, as the detector continues to build up a model of how a series looks every time `detectChangepoints` is called, and uses the existing state.

When I make this change locally I get decent looking results here - sample size of 1, of course :grin:

![image](https://github.com/grafana/explore-logs/assets/5464991/4bec3d60-a619-4cec-891d-995d03be2d4d)

Another plus: it should be faster, since the time/memory complexity is > O(N) in the number of points the detector sees.

Reply:
Ahhhh, I had my suspicions but assumed that instances could be reused. Thanks!